src/test/store/profile-view.test.js

```diff
@@ -116,83 +123,613 @@ describe('call node paths on implementation filter change', function() {
-      [A, E],
-    ]);
+  describe('focusCallTree', function() {
+    const { profile } = getProfileFromTextSamples('A');
+    const { dispatch, getState } = storeWithProfile(profile);
+
+    it('updates the focus call tree generation', function() {
+      expect(
+        ProfileViewSelectors.getFocusCallTreeGeneration(getState())
+      ).toEqual(0);
+      dispatch(ProfileView.focusCallTree());
+      expect(
+        ProfileViewSelectors.getFocusCallTreeGeneration(getState())
+      ).toEqual(1);
+    });
   });
 
-  it('expands subtrees', function() {
+  describe('changeRightClickedThread', function() {
+    const { profile } = getProfileFromTextSamples('A', 'B');
     const { dispatch, getState } = storeWithProfile(profile);
 
-    // First expand A by selecting B
-    dispatch(ProfileView.changeSelectedCallNode(threadIndex, [A, B]));
+    it('changes the right clicked thread index', function() {
+      expect(
+        ProfileViewSelectors.getRightClickedThreadIndex(getState())
+      ).toEqual(0);
+      dispatch(ProfileView.changeRightClickedThread(1));
+      expect(
+        ProfileViewSelectors.getRightClickedThreadIndex(getState())
+      ).toEqual(1);
+    });
+  });
 
-    const callNodeInfo = selectedThreadSelectors.getCallNodeInfo(getState());
+  describe('changeThreadOrder', function() {
+    const { profile } = getProfileFromTextSamples('A', 'B', 'C');
+    const { dispatch, getState } = storeWithProfile(profile);
 
-    // Before expand all action is dispatched, only A is expanded
-    expect(
-      selectedThreadSelectors.getExpandedCallNodePaths(getState())
-    ).toEqual([
-      // Paths
-      [A],
-    ]);
+    it('changes the thread order', function() {
+      expect(UrlStateSelectors.getThreadOrder(getState())).toEqual([0, 1, 2]);
+      withAnalyticsMock(() => {
+        dispatch(ProfileView.changeThreadOrder([2, 1, 0]));
+        expect(self.ga.mock.calls).toEqual([
+          [
+            'send',
+            {
+              eventAction: 'change thread order',
+              eventCategory: 'profile',
+              hitType: 'event',
+            },
+          ],
+        ]);
+      });
+      expect(UrlStateSelectors.getThreadOrder(getState())).toEqual([2, 1, 0]);
+    });
+  });
 
-    dispatch(
-      ProfileView.expandAllCallNodeDescendants(
-        threadIndex,
-        1, // B
-        callNodeInfo
-      )
-    );
+  describe('hideThread', function() {
+    const { profile } = getProfileFromTextSamples('A', 'B', 'C');
+    const { dispatch, getState } = storeWithProfile(profile);
 
-    expect(
-      selectedThreadSelectors.getExpandedCallNodePaths(getState()).sort()
-    ).toEqual([
-      // Paths
-      [A],
-      [A, B],
-      [A, B, C],
-      [A, B, D],
-    ]);
+    it('hides threads', function() {
+      expect(UrlStateSelectors.getHiddenThreads(getState())).toEqual([]);
+      withAnalyticsMock(() => {
+        dispatch(ProfileView.hideThread(1));
+        expect(self.ga.mock.calls).toEqual([
```

Comment:
same

Reply:
I think in the future, if you wouldn't mind, I would prefer a single comment pointing out a fix across the changeset, as I can use my search and replace to fix it. The amount of comments ends up making it harder to navigate the rest of the review. GitHub starts hiding your comments.

Reply:
you're totally right, that's what I usually do and I don't know why I did otherwise here.

Reply:
Thank you sir!